pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/text/LuceneTextIndexReader.java

```diff
@@ -169,5 +178,49 @@ public void close()
       throws IOException {
     _indexReader.close();
     _indexDirectory.close();
+    _docIdReaderWriter.close();
+  }
+
+  private class DocIdReaderWriter implements Closeable {
+    private PinotDataBuffer _buffer;
+
+    DocIdReaderWriter(File segmentIndexDir, String column, int numDocs) throws Exception {
+      int length = Integer.BYTES * numDocs;
+      File docIdMappingFile = new File(SegmentDirectoryPaths.findSegmentDirectory(segmentIndexDir),
+          column + LUCENE_TEXT_INDEX_DOCID_MAPPING_FILE_EXTENSION);
+      // For newly added segments, this file will not exist.
+      // For segment refresh, segment reload and server restart, file will exist,
+      // but we don't know if we are here for refresh v/s reload v/s restart.
+      // In case of refresh, we have to build the mapping again, but in case of
```

Comment:
For segment refresh, this file should not exist as we delete the old segment

Reply:
Yes, that's what I stated in the comment.